linker/shared/src/main/scala/org/scalajs/linker/frontend/BaseLinker.scala

```diff
@@ -107,11 +115,11 @@ final class BaseLinker(config: CommonPhaseConfig, checkIR: Boolean) {
   private def assemble(moduleInitializers: Seq[ModuleInitializer],
       analysis: Analysis)(implicit ec: ExecutionContext): Future[LinkingUnit] = {
     def assembleClass(info: ClassInfo) = {
-      val classAndVersion = irLoader.loadClassDefAndVersion(info.className)
+      val version = irLoader.irFileVersion(info.className)
```

Comment:
To avoid triggering a load if the version stays the same (in other parts of the codebase):

Alternatives:

- (re)-add an additional method `loadClassDefAndVersion` to `IRLoader`. Con: More complicated interface.
- Return `IRFile`s from `IRLoader` (or `Option[IRFile]`). Con: Needs wrapping in the Refiner (OTOH, we could replace the tuples returned by the Optimizer by an `IRFile` subclass).

Reply:
Ah yes, I see. I wouldn't go as far as returning `IRFile` from `IRLoader`. I think I would find that confusing. Either re-add `loadClassDefAndVersion` or leave it as is seem equally good.